src/Microsoft.Azure.WebJobs.Extensions.Storage/Blobs/Listeners/BlobLogListener.cs

```diff
@@ -157,12 +161,20 @@ private static string GetSearchPrefix(string service, DateTime startTime, DateTi
         // Populate the List<> with blob logs for the given prefix. 
         // http://blogs.msdn.com/b/windowsazurestorage/archive/2011/08/03/windows-azure-storage-logging-using-logs-to-track-storage-requests.aspx
         private static async Task GetLogsWithPrefixAsync(List<ICloudBlob> selectedLogs, CloudBlobClient blobClient,
-            string prefix, CancellationToken cancellationToken)
+            string prefix, IWebJobsExceptionHandler exceptionHandler, CancellationToken cancellationToken)
         {
             // List the blobs using the prefix
-            IEnumerable<IListBlobItem> blobs = await blobClient.ListBlobsAsync(prefix,
-                useFlatBlobListing: true, blobListingDetails: BlobListingDetails.Metadata,
-                cancellationToken: cancellationToken);
+            IEnumerable<IListBlobItem> blobs = await DeadlockHandler.RestartIfDeadlocked("ScanLogs", exceptionHandler, () =>
+            {
+                return blobClient.ListBlobsAsync(prefix,
```

Comment:
I didn't realize this was an extension we wrote to loop through all continuation tokens -- I'm going to move this timeout handling code down there...